WordPress/Classes/Services/SiteCreationService.swift

```diff
@@ -216,16 +221,15 @@ open class SiteCreationService: LocalCoreDataService {
                                   success: @escaping (_ blog: Blog) -> Void,
                                   failure: @escaping SiteCreationFailureBlock) {
 
+        status(.creatingSite)
+
         guard let api = account.wordPressComRestApi else {
-            let error = SiteCreationError.missingRESTAPI as NSError
+            let error = SiteCreationError.missingRESTAPI
             DDLogError("Error while creating site: Missing REST API.")
-            assertionFailure()
```

Comment:
The assertions were causing the app to crash. 1) That seemed severe; 2) With the error view these issues are handled gracefully.

Reply:
Yes, the app crashing is intentional, to surface these problems in development. More on this: these assertions only cause crashes in debug builds because the `NS_BLOCK_ASSERTIONS` macro is used for release builds.

We use these kinds of assertions throughout the app (as a project-wide search for `assertionfailure` will show). I think this is a bigger change than should be made as part of this PR; if you want to remove these I think it should be done as a separate PR, and that PR should cover more than the `SiteCreationService.swift`.

Reply:
Ahhhh! Thank you for the insight. It'll be like this never happened! 😉